### PR TITLE
Heroku compat + more

### PR DESCRIPTION
**Pull request by meatballhat**
(Submitting per conversation in #15)

I ran into a few issues when deploying 0.3.0 to Heroku, only one of which is Heroku-specific, afaict:

* generated code in `./node` is not generated with the default buildpack, so I added a `postinstall` hook
* setting `channel` was insufficient for my uses (???) so I settled on this implementation after hacking around a bit and stealing from https://github.com/tech404/inviter
* failed invitations (sometimes?) return a 200, so checking `body.ok` is also needed

### CODE REVIEW

**Comment by mmun** [comment]
Hi @meatballhat! Thanks for putting this together. I tried to "Deploy to Heroku" off your branch but invitations still seem to be failing :(

**Comment by meatballhat** [comment]
I'm also working on supporting TLS for purposes of forcing a redirect, fwiw: https://github.com/meatballhat/slackin/pull/1

File: lib/slack-invite.js [hunk 1]
```diff
@@ -1,25 +1,35 @@
...
   .type('form')
   .send(data)
   .end(function(err, res){
+    console.log(`Invitation request response: err=${err} res.status=${res.status} res.body.ok=${res.body.ok}`);
     if (err) return fn(err);
     if (200 != res.status) {
       fn(new Error(`Invalid response ${res.status}.`));
       return;
     }
+    if (!res.body.ok) {
```
Comment (rauchg) [inline lib/slack-invite.js]: this is the same as the `200` check?

Comment (rauchg) [inline lib/slack-invite.js]: nvm I saw your comment above

File: Procfile [hunk 2]
```diff
@@ -1 +1 @@
-web: make && bin/slackin --port $PORT $SLACK_SUBDOMAIN $SLACK_API_TOKEN
+web: bin/slackin --port $PORT $SLACK_SUBDOMAIN $SLACK_API_TOKEN
```
Comment (rauchg) [inline Procfile]: make is necessary because heroku performs a `git clone`

File: bin/slackin [hunk 3]
```diff
@@ -11,7 +11,7 @@ program
 .option('-c, --channel <chan>', 'Single channel guest invite [$SLACK_CHANNEL]', process.env.SLACK_CHANNEL)
 .option('-i, --interval <int>', 'How frequently (ms) to poll Slack [$SLACK_INTERVAL or 1000]', process.env.SLACK_INTERVAL || 1000)
 .option('-s, --silent', 'Do not print out warns or errors')
-.option('-c, --css <file>', 'Full URL to a custom CSS file to use on the main page')
+.option('-c, --css <file>', 'Full URL to a custom CSS file to use on the main page', process.env.SLACKIN_CSS)
```
Comment (rauchg) [inline bin/slackin]: I like this. We should document it in the `README` too, along the other env vars.

File: lib/slack-invite.js [hunk 4]
```diff
@@ -1,25 +1,35 @@
...
 
 export default function invite({ org, token, email, channel }, fn){
   let data = { email, token };
 
   if (channel) {
     data.channel = channel;
+    data.channels = channel;
     data.ultra_restricted = 1;
     data.set_active = true;
+    data._attempts = 1;
```
Comment (rauchg) [inline lib/slack-invite.js]: nice!

File: lib/slack-invite.js [hunk 5]
```diff
@@ -1,25 +1,35 @@
...
   }
 
+  console.log(`Requesting invite with data=${util.inspect(data)}`);
```
Comment (rauchg) [inline lib/slack-invite.js]: Logging is done differently in the rest of the project. It's event-driven
https://github.com/rauchg/slackin/blob/master/lib/log.js

File: package.json [hunk 6]
```diff
@@ -1,6 +1,9 @@
 {
   "name": "slackin",
   "version": "0.3.1",
+  "scripts": {
+    "postinstall": "make"
```
Comment (rauchg) [inline package.json]: The problem with this is that the NPM version we publish already contains the precompiled source.

**Comment by rauchg** [comment]
Thanks, good stuff!

**Comment by meatballhat** [comment]
I snooze, I lose? :crying_cat_face:

**Comment by rauchg** [comment]
I thought we'd addressed these issues with other PRs?

**Comment by rauchg** [comment]
😨

**Comment by meatballhat** [comment]
@rauchg works for me! :heart: :heart: :heart:

**Pull request closed without merging**